From d1130d3d666f45af6bacdfb56a7286016df78d3e Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Sat, 25 Feb 2023 13:06:22 +0900
Subject: [PATCH] [ozone/wayland] Don't call eglSwapBuffers() when window isn't
 visible

It causes crash or dead lock with --in-process-gpu.

Signed-off-by: Takuro Ashie <ashie@clear-code.com>
---
 ui/ozone/platform/wayland/gpu/gl_surface_wayland.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ui/ozone/platform/wayland/gpu/gl_surface_wayland.cc b/ui/ozone/platform/wayland/gpu/gl_surface_wayland.cc
index b428f74401c9f..b8eb6be674481 100644
--- a/ui/ozone/platform/wayland/gpu/gl_surface_wayland.cc
+++ b/ui/ozone/platform/wayland/gpu/gl_surface_wayland.cc
@@ -78,6 +78,9 @@ EGLConfig GLSurfaceWayland::GetConfig() {
 }
 
 gfx::SwapResult GLSurfaceWayland::SwapBuffers(PresentationCallback callback) {
+  if (!window_->IsVisible())
+    return gfx::SwapResult::SWAP_FAILED;
+
   UpdateVisualSize();
   if (!window_->IsSurfaceConfigured()) {
     // The presentation |callback| must be called after gfx::SwapResult is sent.
-- 
2.34.1

